docs/design/autoware-architecture/planning/index.md

```diff
@@ -2,43 +2,120 @@
 
 ## Overview
 
-Planning stack acts as the “brain” of autonomous driving. It uses all the results from Localization, Perception, and Map stacks to decide its maneuver and gives final trajectory to Control stack.
+The Planning component generates the trajectory message that will be subscribed to by the Control component based on the environmental state obtained from the Localization and the Perception components. 
+
+## Requirements
+
+The goal of the Planning component is to generate a trajectory (path and velocity) of the ego vehicle that is safe and well-regulated while satisfying the given mission.
+
+**Goals:**
+ - The basic functions are provided so that a simple ODD can be defined.
+ - The functionality is modularized to accommodate the third-party components. That is, a complicated or realistic ODD needs not be defined by the basic functions provided by default.
+ - The capability is extensible with the third-party components or the decision of human operators.
+ - The mechanism and policy are separated to allow the system or operators to change the behavior of the ego vehicle. Ultimately speaking, the policy can be set to crash an obstacle and the mechanism always follows. Otherwise, the system is not safe from the design point of view.
+
+**Non-goals:**
+ - The Planning component is not self-contained but can be extended with third parties.
+ - The Planning component is not aimed at the complete functionality and capability.
+ - The Planning component is not designed to always outperform human drivers.
+ - The Planning component is not capable of “never crashes”.
+
+## High-level architecture
+
+This diagram describes the high-level architecture of the Planning Component.
+
+![overall-planning-architecture](image/high-level-planning-diagram.drawio.svg)
+
+The Planning component consists of the following components:
```

Comment:
(minor) if possible it is nicer to avoid repeating "component" twice.
How about using "functionalities" or "submodules" instead ?
```suggestion
The Planning component consists of the following functionalities:
```

Reply:
@maxime-clem I see. how about sub-components?

For some reason, I don't want to use the "functionality" and "module". They will be used for another meaning later.

Reply:
"Sub-components" is also good :smiley:

Reply:
Done in https://github.com/autowarefoundation/autoware-documentation/pull/355/commits/ff43a54470f6830bfb2f61a5b2eaf067fe5209fa !